bin/service-mango-health/src/processors/persister.rs

```diff
@@ -0,0 +1,725 @@
+use crate::configuration::Configuration;
+use crate::processors::health::{HealthComponent, HealthEvent};
+use anchor_lang::prelude::Pubkey;
+use chrono::{Duration, DurationRound, Utc};
+use fixed::types::I80F48;
+use futures_util::pin_mut;
+use postgres_types::{ToSql, Type};
+use services_mango_lib::fail_or_retry;
+use services_mango_lib::postgres_configuration::PostgresConfiguration;
+use services_mango_lib::postgres_connection;
+use services_mango_lib::retry_counter::RetryCounter;
+use std::collections::{HashMap, VecDeque};
+use std::str::FromStr;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use tokio::task::JoinHandle;
+use tokio_postgres::binary_copy::BinaryCopyInWriter;
+use tokio_postgres::{Client, Transaction};
+use tracing::{error, warn};
+
+pub struct PersisterProcessor {
+    pub job: JoinHandle<()>,
+}
+
+impl PersisterProcessor {
+    pub async fn init(
+        data_sender: &tokio::sync::broadcast::Sender<HealthEvent>,
+        configuration: &Configuration,
+        exit: Arc<AtomicBool>,
+    ) -> anyhow::Result<PersisterProcessor> {
+        let mut data = data_sender.subscribe();
+        let postgres_configuration = configuration.postgres.clone().unwrap_or_default();
+        let persistence_configuration = configuration.persistence_configuration.clone();
+        let time_to_live = Duration::seconds(persistence_configuration.history_time_to_live_secs);
+        let periodicity = Duration::seconds(persistence_configuration.persist_max_periodicity_secs);
+        let max_snapshot_count = persistence_configuration.snapshot_queue_length;
+
+        let mut unpersisted_snapshots = VecDeque::new();
+
+        let job = tokio::spawn(async move {
+            if !persistence_configuration.enabled {
+                return;
```

Comment:
This job exits immediately, but the logger should still keep going. 
Broadcast are allowed to any number of receivers (and it can change during run time)